src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,183 @@
+        user.setPassword(DEFAULT_PASSWORD);
+        registrationService.register(user);
+        assertEquals(user, storageDao.get(user.getLogin()));
+    }
+
+    @Test
+    void register_validAge_ok() {
+        user.setAge(DEFAULT_AGE);
+        registrationService.register(user);
+        assertEquals(user, storageDao.get(user.getLogin()));
+    }
+
+    @Test
+    void register_duplicateUser_notOk() {
+        registrationService.register(user);
+
+        User duplicateUser = new User();
+        duplicateUser.setLogin(DEFAULT_LOGIN);
+        duplicateUser.setPassword(DEFAULT_PASSWORD);
+        duplicateUser.setAge(DEFAULT_AGE);
+
+        assertThrows(
+                InvalidUserDataException.class,
+                () -> registrationService.register(duplicateUser),
+                "Expected register() to throw exception, but it didn't"
+        );
+
+        int userCount = 0;
+        for (User user : Storage.people) {
+            if (user.getLogin().equals(DEFAULT_LOGIN)) {
+                userCount++;
+            }
+        }
+        assertEquals(1, userCount, "Duplicate user should not be registered");
+    }
+
+    @Test
+    void register_loginLessThanSixCharacters_notOk() {
+        user.setLogin(LESS_THAN_6_CHARACTERS);
+        assertThrows(InvalidUserDataException.class, () -> registrationService.register(user));
+        assertNull(storageDao.get(user.getLogin()));
+    }
+
+    @Test
+    void register_passwordLessThanSixCharacters_notOk() {
+        user.setPassword(LESS_THAN_6_CHARACTERS);
+        assertThrows(InvalidUserDataException.class, () -> registrationService.register(user));
+        assertNull(storageDao.get(user.getLogin()));
+    }
+
+    @Test
+    void register_ageLessThan18_notOk() {
+        user.setAge(UNDERAGE);
+        assertThrows(InvalidUserDataException.class, () -> registrationService.register(user));
+        assertNull(storageDao.get(user.getLogin()));
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        user.setLogin(NULL_STRING);
+        assertThrows(InvalidUserDataException.class, () -> registrationService.register(user));
+        assertNull(storageDao.get(user.getLogin()));
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        user.setPassword(NULL_STRING);
+        assertThrows(InvalidUserDataException.class, () -> registrationService.register(user));
+        assertNull(storageDao.get(user.getLogin()));
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        user.setAge(NULL_INTEGER);
+        assertThrows(InvalidUserDataException.class, () -> registrationService.register(user));
+        assertNull(storageDao.get(user.getLogin()));
+    }
+
+    @Test
+    void register_ageZero_notOk() {
+        user.setAge(ZERO_AGE);
+        assertThrows(InvalidUserDataException.class, () -> registrationService.register(user));
+        assertNull(storageDao.get(user.getLogin()));
+    }
+
+    @Test
+    void register_loginExactlySixCharacters_ok() {
```

Comment:
not a mistake, but just my opinion - when you change the limit from 6 to 8, you may forget to change method names, and they will become misleading, so better name it neutrally like tooShort